apps/admin_app/lib/admin_app_web/controllers/option_type_controller.ex

```diff
@@ -0,0 +1,61 @@
+defmodule AdminAppWeb.OptionTypeController do
+  use AdminAppWeb, :controller
+  alias Snitch.Data.Model.OptionType, as: OTModel
+  alias Snitch.Data.Schema.OptionType, as: OTSchema
+
+  def index(conn, _params) do
+    option_types = OTModel.get_all()
+    render(conn, "index.html", %{option_types: option_types})
+  end
+
+  def new(conn, _params) do
+    changeset = OTSchema.create_changeset(%OTSchema{}, %{})
+    render(conn, "new.html", changeset: changeset)
+  end
+
+  def create(conn, %{"option_type" => params}) do
+    case OTModel.create(params) do
+      {:ok, _} ->
+        option_types = OTModel.get_all()
+        render(conn, "index.html", %{option_types: option_types})
+
+      {:error, changeset} ->
+        render(conn, "new.html", changeset: %{changeset | action: :new})
+    end
+  end
+
+  def edit(conn, %{"id" => id}) do
+    with {id, _} <- Integer.parse(id),
+         option_type <- OTModel.get(id) do
```

Comment:
This is dangerous!
What if the `id` is junk? Please handle the error case.
(same problem with `update/2` and `delete/2`)

Reply:
Please test this properly! I can confirm that the `:error` clause never matches, because when the `:id` is junk, the Repo returns `nil`, not `:error`.
To reproduce this error, create a new option type, and let's say the id is `N`.
Try to hit this URL:
```elixir
"http://0.0.0.0:4000/option_types/#{N+1}/edit"
```